src/os/color.js

```diff
@@ -740,6 +740,37 @@ os.color.colorize = function(data, tgtColor) {
   }
 };
 
+/**
+ * Applies a color transform to an array of image data. This transform takes a target color and adjusts the color
+ * for saturation color and brightness.
+ * @param {Array<number>} data The image data to colorize
+ * @param {number} brightness The target brightness. The range is -1 to 1.
+ * @param {number} contrast The target contrast. The range is 0 to 2.
+ * @param {number} saturation The target saturation. The range is 0 to 1.
+ */
+os.color.adjustColor = function(data, brightness, contrast, saturation) {
+  var intercept = (1 - contrast) / 2;
+  var sr = (1 - saturation) * 0.3086;
+  var sg = (1 - saturation) * 0.6094;
+  var sb = (1 - saturation) * 0.0820;
+  brightness = brightness * 225;
+
+  // color transform matrix for contrast and saturation, taken from online research
+  var m = [
+    contrast * (sr + saturation), contrast * sr, contrast * sr, 0, 0,
+    contrast * sg, contrast * (sg + saturation), sg * contrast, 0, 0,
+    contrast * sb, contrast * sb, contrast * (sb + saturation), 0, 0,
+    0, 0, 0, 1, 0,
+    intercept + brightness, intercept + brightness, intercept + brightness, 1, 0
+  ];
+
+
+  for (var i = 0, n = data.length; i < n; i += 4) {
+    data[i] = (data[i] * m[0] + data[i + 1] * m[5] + data[i + 2] * m[10] + data[i + 3] * m[15] + m[20]);
+    data[i + 1] = (data[i] * m[1] + data[i + 1] * m[6] + data[i + 2] * m[11] + data[i + 3] * m[16] + m[21]);
```

Comment:
Is `data[i]` on the right side of this expression intended to be the already-recalculated `data[i]` or the original `data[i]`?

Same for that and also `data[i + 1]` on the next line.

Reply:
It should probably be the original I will fix that.